### DISCUSSION TITLE
Printing size for best readably for scanned docs

### DISCUSSION THREAD

**Discussion by mPisano** [micjahn/ZXing.Net#432] [Q&A]
First Fantastic project and excellent work!

I'm using a CODE 128 barcode printed on a delivery slip that is later scanned back in.
Truckers aren't that neat, so they get dirty & crumpled and then they are scanned back (typically Xerox) at 200 or 300 dpi (sometimes slightly skewed).
I then try to decode the barcode from the scan and link it back to the order.

It's working, any my success rate is around 90%, but I thought I would ask is there a better or "Proper" way to scale a barcode for better detection? 

                        Format = ZXing.BarcodeFormat.CODE_128,
                        Options = new ZXing.Common.EncodingOptions
                        {
                            PureBarcode = true,
                            Height = 75,
                            Width = 1300 
                        }
                        Bitmap BMbcode = ZXbcode.Write("ABC1234567")

My theory was larger is better especially at 200 dpi, my barcodes are about an 1 x 3 on the paper and I draw a thick box around it with a decent "clear zone" in between.

Also - are any of the other formats (even 2d) more resilient and better for this task?

Appreciate the thoughts, knowledge  and direction :)

Thanks,
Mike

**Comment by micjahn** [comment]
I don't think that there is one way (or size) that fits all needs. You should try different sizes in my opinion. Bigger isn't better in all cases, especially if the code should be scanned by mobile devices or barcode scanner handhelds by a short distance. I would start with a size which is normally used on consumer products. Increase the size by small steps. Measure the scanning success rate. Perhaps you can print two barcodes on the page in different sizes at different places. If one of them fails you have a second try.

**Comment by mPisano** [comment]
Thanks for the response.

In my use case, the delivery slip is the product.
ZXing is creating 10 digit alpha barcode, its printed, signed, scanned back in and ZXing is detecting.

For the oned detection with try harder I though a one inch in height would be enough, was looking for more direction on barcode width and if a CODE 128 is best for this application

I will try different sizes, and a second barcode, my biggest issues so far are 200 dpi copiers and the dithering pattern confusing detection.

Thanks again,
Mike
